jfx_bridge/bridge.py

```diff
@@ -1163,7 +1169,10 @@ def __init__(self, bridge_conn, obj_dict):
         self._bridge_overrides = dict()
 
     def __getattribute__(self, attr):
-        if attr.startswith(BRIDGE_PREFIX) or attr == "__class__" or attr in BridgedObject._DONT_BRIDGE or attr in BridgedObject._LOCAL_METHODS or (attr in BridgedObject._DONT_BRIDGE_UNLESS_IN_ATTRS and attr not in self._bridge_attrs):
+
+        if attr in _EXTERNAL_OVERRIDES:
```

Comment:
Per the comment above, this would need to change to getting _EXTERNAL_OVERRIDES from the self.\_bridge_conn object.

That in turn means a little bit of playing around with the ordering of the attribute resolution - `if attr.startswith(BRIDGE_PREFIX)` would need to be pulled out separately so it always runs first, so you can get the self.\_bridge_conn without going into a loop.

This means you wouldn't be able to override any of the \_bridge_ prefix attributes - but I think that's probably actually okay. I like the idea of avoiding people breaking core bridge functionality with overrides.

Reply:
In my opinion this API should come with a giant warning that you should really now what you are doing before using it anyway. You could probably still end up with some loops if your override accesses some attribute that is also overridden. Probably as soon as some attribute is a reference to the object itself, but that is the issue of the override IMO.

But not allowing to overwrite `BRIDGE_PREFIX` attributes seems sensible, to make this cleaner. The core goal of this is to make the object that is being proxied more convincing to any attempts of introspection, so `BRIDGE_PREFIX` attributes are irrelevant for that. For example I would still want to override attributes like `__class__` . The override could involve hacky stack frame introspection to return a different object if IPython is asking, and just concede for anything else.

Will change this accordingly.